korangar/src/main.rs

```diff
@@ -2384,6 +2395,12 @@ impl ApplicationHandler for Client {
                 if !focused {
                     self.input_system.reset();
                     self.focus_state.remove_focus();
+                    let mute_on_focus_loss = *self.mute_on_focus_loss.get();
+                    if mute_on_focus_loss {
+                        self.audio_engine.mute(true);
+                    }
+                } else {
+                    self.audio_engine.mute(false);
```

Comment:
You should only unmute it if "mute_on_focus_loss" is set!

So better move the mute function out of the `if !focused` and instead write:

```rust
let mute_on_focus_loss = *self.mute_on_focus_loss.get();
if mute_on_focus_loss {
    self.audio_engine.mute(!focused);
}
```

```

Reply:
If you mute with mute_on_focus_loss activated and later you deactivate mute_on_focus_loss during the mute, the audio will stay muted.

Reply:
But here we handle the case where the window if focuses again. You can't change the "mute_on_focus_loss" option while you are unfocused!

Reply:
True. Theoretically it is possible.